### PR TITLE
Team E: Leonard, Vickie, Brendan

### PR DESCRIPTION
**Pull request by vippie852**
http://htmlpreview.github.io/?https://github.com/vippie852/PS02/blob/master/PS02.html

### CODE REVIEW

File: PS02.Rmd [hunk 1]
```diff
@@ -10,36 +10,112 @@ output:
...
+train %>% 
+  ggplot(aes(x = "", y = log_price)) + geom_boxplot(color="darkred") + theme_bw() + coord_flip() + labs(title = "Log Sale Price in Training Data", x = "", y = "Log Sale Price (Log $)")
+favstats(train$log_price)
 ```
+The transformation helped to transform `price_doc` into a bell-shaped distribution. This is shown by the symmetry in the histogram. Note that outliers still exist and are easily spotted in the boxplot above.
 
-
-# Model Fit
-
+## Model Fit
```
Comment (akim17) [inline PS02.Rmd]: Even though this might be a good way to determine which explanatory variable would be the best in terms of predicting the price, the ultimate goal of the assignment is to create a spline model with an explanatory variable that you think would be most helpful in predicting the price. Applying the spline should actually be the last part of the model fitting process. As a suggestion, other groups used correlation plots and linear regression to find the best continuous numeric explanatory variable for their spline. Consider these alternative methods when picking `x`.

Also, you need to determine what the best degrees of freedom is for your model. Why is 10 picked? Why not something like 7 or 9? You should do a comparison of different degrees of freedom to justify the number you picked.

File: PS02.Rmd [hunk 2]
```diff
@@ -10,36 +10,112 @@ output:
...
+  # Compare predicted price with actual price.  Sum total RMSLE
+  combined <- data_frame(id = newX$id, price_new = output$y, price_old = newX$price_doc)
+  combined <- combined %>% mutate(SE = (log(price_new+1)-log(price_old+1))^2) %>% summarise(RMSE = sqrt(mean(SE)), n = n(), variable = var) # root mean squared log error 
+  # New data frame contains variable name and score
+  return(combined)
+}
+```
+Next, we ranked all the continuous integer variables based on their RMSLE's and found out that `full_sq` had the lowest RMSLE!
 ```{r}
+scores <- sapply(clas$name[c(1:22, 38:47, 51:55,58:71, 73:76, 78:82, 84:122)], spleen)
```
Comment (akim17) [inline PS02.Rmd]: I'm getting a compilation error here. Here's what I got:

```
Error: 'as.tibble' is not an exported object from 'namespace:tibble'
```

File: PS02.Rmd [hunk 3]
```diff
@@ -10,36 +10,112 @@ output:
...
+```{r, warning=FALSE}
+train <- train %>%
+  mutate(
+    log_full = log(full_sq)
+  )
+train %>% 
+  ggplot(aes(x = "", y = log_full)) + geom_boxplot() + theme_bw() + coord_flip() + labs(title = "Property's Total Log Area from Training Data", x = "", y = "Log Total Area (log square meters)")
 ```
 
+```{r, include = FALSE}
```
Comment (akim17) [inline PS02.Rmd]: I would suggest taking out the `include = FALSE` and have this code show. I would put a header on this section called "Submission"

File: PS02.Rmd [hunk 4]
```diff
@@ -10,36 +10,112 @@ output:
...
+# Create Submission File
+mspline <- with(train1, smooth.spline(full_sq, price_doc, df=10))
+newX <- test %>%
+  filter(!is.na(full_sq))%>%select(full_sq)
+output <- predict(mspline, newX$full_sq) %>%
```
Comment (akim17) [inline PS02.Rmd]: I'm also getting another error here too:

```
Error: 'as.tibble' is not an exported object from 'namespace:tibble'
```

File: PS02.Rmd [hunk 5]
```diff
@@ -10,36 +10,117 @@ output:
...
+## Introduction
+For Problem Set #2, our assignment was to enter the Sberbank Russian Housing Market Kaggle competition and attempt to fit a spline model to predict the sale price of a piece of Russian real estate, `price_doc`. 
 
-# Exploratory Data Analysis
-
+## Exploratory Data Analysis (Response Variable)
+First off, we'd like to look into our response variable in the training set, `price_doc`:
+```{r} 
+train %>% 
+  ggplot(aes(x = "", y = price_doc)) + geom_boxplot() + theme_bw() + coord_flip() + labs(title = "Sale Price in Training Data", x = "", y = "Sale Price ($)")
```
Comment (rudeboybert) [inline PS02.Rmd]: Coding style nitpick: consider having each `+` layer on its own line; that way its easier to see all the elements of your ggplot.

File: PS02.Rmd [hunk 6]
```diff
@@ -10,36 +10,117 @@ output:
...
+test <- read_csv("data/test.csv")
 train <- read_csv("data/train.csv")
 ```
 
+## Introduction
+For Problem Set #2, our assignment was to enter the Sberbank Russian Housing Market Kaggle competition and attempt to fit a spline model to predict the sale price of a piece of Russian real estate, `price_doc`. 
 
-# Exploratory Data Analysis
-
+## Exploratory Data Analysis (Response Variable)
```
Comment (rudeboybert) [inline PS02.Rmd]: put blank lines between the ## section names so that the section heads show up in HTML file

File: PS02.Rmd [hunk 7]
```diff
@@ -10,36 +10,117 @@ output:
...
+  combined <- combined %>% 
+    mutate(SE = (log(price_new+1)-log(price_old+1))^2) %>% # root mean squared log error
+    summarise(RMSE = sqrt(mean(SE)), n = n(), variable = var, df = mspline$df)
+  # New data frame contains variable name, score and optimal df
+  return(combined)
+}
+```
+Next, we ranked all the continuous integer variables based on their RMSLE's and found out that `full_sq` had the lowest RMSLE!
+```{r}
+scores <- sapply(clas$name[c(1:22, 38:47, 51:55,58:71, 73:76, 78:82, 84:122)], spleen)
```
Comment (rudeboybert) [inline PS02.Rmd]: you can not have warning messages show up by setting `warning=FALSE` in the R code chunk header.

File: PS02.Rmd [hunk 8]
```diff
@@ -10,36 +10,117 @@ output:
...
+train %>% 
+  ggplot(aes(x = "", y = log_full)) + geom_boxplot() + theme_bw() + coord_flip() + labs(title = "Property's Total Log Area from Training Data", x = "", y = "Log Total Area (log square meters)")
 ```
 
-
-# Create Submission File
-
+## Model Fit
+In order to find the variable that best predicts `price_doc`, we decided to create a function that ranks the variables based on their root mean squared log errors (RMSLE). The variable with the lowest RMSLE value will become the explanatory variable of our model.
+First, to create a model that avoids the Texas Shooter Fallacy, we decided to resample from the training set. We then removed data from the test set that had been resampled using the `anti_join` function.
```
Comment (rudeboybert) [inline PS02.Rmd]: Good call here! However, I wouldn't phrase this in a report as "the Texas SSF", as that's more of an analogy I drew. Rather, something like "not using the same data to train/fit model and test its performance" or "to avoid overfitting" is slightly more appropriate language in this case.
